### DISCUSSION TITLE
Redirecting a dynamic route to outer domain using redirects of next.config

### DISCUSSION THREAD

**Discussion by sushantgwr87** [vercel/next.js#60370] [Help] [answered]
### Summary

Currently working with **Next.js 12.2.3** version in a large project which has blogs created via [slug], dynamic routes basically using getStaticProps.

Now lets say this project domain is: `domain-1.com`
and blog url is: `domain-1.com/blogs/going-to-redirect`

Now I am moving some Page urls to an outer domain, mainly blogs and 2 page. Some Buttons and links of these urls are present on pages which **I am not redirecting** so if user clicks on those link it should go to these redirected pages.

Now let say new outer domain is: `domain-2.com`
and new blog route is: `domain-2.com/blog/redirected-blog`

For page routes, the clicking on old links redirect them to new outer page but when user click on these blog links, it fails. First it shows application error:

### Application error: a client-side exception has occurred (see the browser console for more information).

On refreshing this page with error, it redirects to correct page. How to fix this so that those links redirect directly on clicking old links.

### Sample link:
```
<Link href="/blogs/going-to-redirect">
    read Blog
</Link>
```

### Next.config redirect:
```
{
    source: '/blogs/going-to-redirect',
    destination: 'https://domain-2.com/blog/redirected-blog',
    permanent: true
  },
```

**Comment by icyJoseph** [comment, marked as answer]
I wonder if the issue is that props are expected for those routes, but since they are getting redirected instead, the page crashes?

I also wonder if it maybe is just a bug that was fixed on a later version. Could you try to upgrade a couple of minor versions ahead?

**Reply by sushantgwr87** [reply]
> I wonder if the issue is that props are expected for those routes, but since they are getting redirected instead, the page crashes?

Yes, I thought same and found that it is a problem. After checking console logs I found same thing, getStaticProps fetches the data as it still exist there but is not able to render it because URL gets redirected.

Well I upgraded my next version from 12 to 13.5 and it seems it has fixed this issue. Now on Clicking the link with old urls it takes 100ms (even less) and correctly redirects to new url instead of crashing it.

I think upgrading to latest version or upto a stable version of packages fixes most of such minor issues. This project is way old but was working fine until we decided this redirection. Though now I need to upgrade all components as well according Next 13 since it was kind off major update in their base Components of Next-Image and Next-links.

Thanks for helping. 🥂
